### ISSUE TITLE
Why is the Event state parameter "exclusive" called that?

### ISSUE THREAD

**Issue by LCaparelli** [serverlessworkflow/specification#160]
**What is the question**:

I understand what the parameter does, but I'm not sure why it was called that. :-)

[Event state definition](https://github.com/serverlessworkflow/specification/blob/master/specification.md#event-state).

**Comment by tsurdilo** [comment]
@LCaparelli exclusive meaning "just one, not admitting others". Imo its fitting of what its supposed to represent but please recommend other possible naming so we can consider.

**Comment by LCaparelli** [comment]
@tsurdilo hmmm, "exclusive" kind of gives me the impression that it will only run if only one of the events arrive (as in XOR). While in reality, we want to express a situation where all conditions must be met before proceeding (as in AND).

If understand the defined behavior correctly, if `exclusive` is set to `false`, it will promptly execute the actions bound to a particular event that arrives while if set to `true`, it waits for all events before executing the actions. This reminds of eager vs. lazy type of scenarios, so maybe this could be named `lazy` instead. Wdyt?

Alternatively, we continue with the logical gates analogy and replace this parameter with a `kind` parameter, which should accept one of:

  - XOR: only execute if a single event arrived within the timeout (this brings a certain degree of complexity, `timeout` becomes required for this kind)
  - AND: only execute if all events arrive
  - OR: execute if any event arrive

Not sure I like this any better, I'm uncertain if XOR would be worth the trouble or if it even is a useful behavior, but would love to hear your thoughts on it.

**Comment by tsurdilo** [comment]
@LCaparelli your first statement is actually correct 

I'll try to give example:

```json
"onEvents": [{
        "eventRefs": ["EventA", "EventB"],
        "actions": [{
            "functionRef": {
                "refName": "someFunctionDefName"
            }
        }]
    }]
```
if exclusive is set to true then either EventA OR EventB can trigger the action
if exclusive is set to false then EventA AND EventB trigger action

**Comment by LCaparelli** [comment]
> @LCaparelli your first statement is actually correct

Well, not quite. My first statement was closer to:

if exclusive is set to true then only EventA OR only EventB can trigger the action.

Like in an XOR logical gate, if more than one input is true, the outcome is false. So if **both** EventA and EventB arrive, no actions are triggered.

This is totally subjective of course, different people will understand it differently, but that's the first impression I got when saw the `exclusive` name. :-)

I'm not fully satisfied with the suggestions I made either, so I'm not against keeping it as is and if more people find it confusing in the future we may revisit this.
